tests/bwrite.rs

```diff
@@ -17,128 +17,554 @@
+    ])
+    .output()?;
+
+    let mut c = tm.command()?;
+    let scan_cmd = c.args(&[
         "--region",
         "local",
         "--table",
         &table_name,
+        "scan",
+        "-o",
+        "raw",
+    ]);
+    scan_cmd.assert().success().stdout("[]\n");
+
+    let mut c = tm.command()?;
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name_sk,
+        "scan",
+        "-o",
+        "raw",
+    ]);
+    scan_cmd.assert().success().stdout("[]\n");
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_batch_write_json_put_delete() -> Result<(), Box<dyn std::error::Error>> {
+    let mut tm = util::setup().await?;
+    let table_name = tm
+        .create_temporary_table_with_items(
+            "pk",
+            None,
+            vec![util::TemporaryItem::new(
+                "ichi",
+                None,
+                Some(r#"{"null-field": null}"#),
+            )],
+        )
+        .await?;
+
+    let tmpdir = Builder::new().tempdir()?; // defining stand alone variable here as tempfile::tempdir creates directory and deletes it when the destructor is run.
+    let batch_input_file_path = create_test_json_file(
+        "tests/resources/test_batch_write_put_delete.json",
+        vec![&table_name],
+        &tmpdir,
+    );
+
+    let mut c = tm.command()?;
+    c.args(&[
+        "--region",
+        "local",
         "bwrite",
         "--input",
-        batch_input_file_path.to_str().unwrap(),
+        &batch_input_file_path,
     ])
     .output()?;
 
     let mut c = tm.command()?;
-    let scan_cmd = c.args(&["--region", "local", "--table", &table_name, "scan"]);
-    scan_cmd
-        .assert()
-        .success()
-        .stdout(predicate::str::is_match("pk *attributes\nichi").unwrap());
-
-    /*
-    get output should looks like:
-        $ dy --region local -t table--test_batch_write get ichi
-        {
-          "Dimensions": [
-            "Giraffe",
-            "Hippo",
-            "Zebra"
-          ],
-          "PageCount": [
-            -19.0,
-            3.14,
-            7.5,
-            42.2
-          ],
-          "Authors": [
-            "Author1",
-            "Author2",
-            42
-          ],
-          "InPublication": false,
-          "Nothing": null,
-          "Price": 2,
-          "pk": "ichi",
-          "Details": {
-            "Age": 35,
-            "Misc": {
-              "dream": [
-                35,
-                null
-              ],
-              "hope": true
-            },
-            "Name": "Joe"
-          },
-          "ISBN": "111-1111111111"
-        }
-    */
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "scan",
+        "-o",
+        "json",
+    ]);
+    scan_cmd.assert().success().stdout(
+        predicate::str::is_match(r#"pk": "ni""#)?.and(predicate::str::is_match(r#"pk": "san""#)?),
```

Comment:
I believe it must be `r#""pk": "ni""#` and `r#""pk": "san""`.

Reply:
You are right. Fixed.